File: py/portfolio.py
import functools
import operator
from money import Money

class Portfolio:
    def add(self, *moneys):
        self.moneys.extend(moneys)
    
    def evaluate(self, currency):
        total = functools.reduce(
            operator.add, map(lambda m: m.amount, self.moneys)
        , 0)
        return Money(total, currency)
    
    def __init__(self):
        self.moneys = []

File: js/portfolio.js
const Money = require('./money');

class Portifolio {
    constructor(){this.moneys = []}
    add(){
        this.moneys =
        this.moneys.concat(Array.prototype.slice.call(arguments));
    }
    evaluate(currency){
        let total = this.moneys.reduce((sum, money) => {
            return sum + this.convert(money, currency);
        }, 0);
        return new Money(total, currency);
    }
    convert(money, currency){
        let eurToUsd = 1.2;
        if(money.currency === currency){
            return money.amount;
        }
        return money.amount * eurToUsd;
    }
}

module.exports = Portifolio;